utils/zerovec/src/varzerovec/lengthless.rs

```diff
@@ -0,0 +1,132 @@
+    ///
+    /// # Safety
+    ///
+    /// `bytes` need to be an output from [`VarZeroLengthlessSlice::as_bytes()`], or alternatively
+    /// be valid to be passed to `from_bytes_unchecked_with_length`
+    ///
+    /// The length associated with this value will be the length associated with the original slice.
+    pub(crate) const unsafe fn from_bytes_unchecked(bytes: &[u8]) -> &Self {
+        // self is really just a wrapper around a byte slice
+        mem::transmute(bytes)
+    }
+
+    /// Uses a `&mut [u8]` buffer as a `VarZeroLengthlessSlice<T>` without any verification.
+    ///
+    /// # Safety
+    ///
+    /// `bytes` need to be an output from [`VarZeroLengthlessSlice::as_bytes()`], or alternatively
+    /// be valid to be passed to `from_bytes_unchecked_with_length`
+    ///
+    /// The length associated with this value will be the length associated with the original slice.
+    pub(crate) unsafe fn from_bytes_unchecked_mut(bytes: &mut [u8]) -> &mut Self {
+        // self is really just a wrapper around a byte slice
+        mem::transmute(bytes)
+    }
+
+    /// Get one of this slice's elements
+    ///
+    /// # Safety
+    ///
+    /// `index` must be in range, and `len` must be the length associated with this
+    /// instance of VarZeroLengthlessSlice.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # use zerovec::VarZeroVec;
+    ///
+    /// let strings = vec!["foo", "bar", "baz", "quux"];
+    /// let vec = VarZeroVec::<str>::from(&strings);
+    ///
+    /// let mut iter_results: Vec<&str> = vec.iter().collect();
+    /// unsafe {
+    ///     assert_eq!(vec.get_unchecked(0), "foo");
+    ///     assert_eq!(vec.get_unchecked(1), "bar");
+    ///     assert_eq!(vec.get_unchecked(2), "baz");
+    ///     assert_eq!(vec.get_unchecked(3), "quux");
+    /// }
+    /// ```
+    pub(crate) unsafe fn get_unchecked(&self, len: u32, idx: usize) -> &T {
+        self.as_components(len).get_unchecked(idx)
+    }
+
+    /// Get a reference to the entire encoded backing buffer of this slice
+    ///
+    /// The bytes can be passed back to [`Self::parse_byte_slice()`].
+    ///
+    /// To take the bytes as a vector, see [`VarZeroVec::into_bytes()`].
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # use zerovec::VarZeroVec;
+    ///
+    /// let strings = vec!["foo", "bar", "baz"];
+    /// let vzv = VarZeroVec::<str>::from(&strings);
+    ///
+    /// assert_eq!(vzv, VarZeroVec::parse_byte_slice(vzv.as_bytes()).unwrap());
+    /// ```
+    #[inline]
+    pub(crate) const fn as_bytes(&self) -> &[u8] {
+        &self.entire_slice
+    }
+
+    pub(crate) unsafe fn get_bytes_at_mut(&mut self, len: u32, idx: usize) -> &mut [u8] {
+        let components = self.as_components(len);
+        let range = components.get_things_range(idx);
+        let offset = components.get_indices_size();
+
+        #[allow(clippy::indexing_slicing)] // get_range() is known to return in-bounds ranges,
```

Comment:
Nit: This comment looks old

```suggestion
         // get_indices_size() returns the start of the things slice, and get_things_range()
         // returns a range in-bounds of the things slice
        #[allow(clippy::indexing_slicing)]
```